app/plugin_install.go

```diff
@@ -413,21 +419,81 @@ func (a *App) removePlugin(id string) *model.AppError {
 	if err = a.removeSignature(id); err != nil {
 		mlog.Error("Can't remove signature", mlog.Err(err))
 	}
-
-	a.notifyClusterPluginEvent(
-		model.CLUSTER_EVENT_REMOVE_PLUGIN,
-		model.PluginEventData{
-			Id: id,
-		},
-	)
+	canaryTag := *a.Config().PluginSettings.CanaryTag
+	if canaryTag == "" {
+		a.notifyClusterPluginEvent(
+			model.CLUSTER_EVENT_REMOVE_PLUGIN,
+			model.PluginEventData{
+				Id: id,
+			},
+		)
+	}
 
 	if err := a.notifyPluginStatusesChanged(); err != nil {
 		mlog.Error("Failed to notify plugin status changed", mlog.Err(err))
 	}
 
+	if canaryTag == "" {
```

Comment:
It's been a while, so I've forgotten the details here, but it seems like this block could use some clarifying documentation. Something akin to:
```
If no canary tag is configured, remove all instances, including ones installed from servers with a configured canary tag.
Otherwise, if a canary tag is configured, remove the one installed with the canary tag, and try to re-install one without, if it exists.
```

Writing this down leads me to ask: what happens when I try to uninstall /that/ plugin from a server with a canary tag configured? i.e., can I uninstall the untagged plugin from that server?

Reply:
@lieut-data, you can not remove plugin from the server where canary tag is configured. You have to remove it from the server without a canary tag. I'm 0/5 if that should be the case though.